app/models/attachment.rb

```diff
@@ -297,13 +277,27 @@ def delete_file_and_thumbnail
     attachment_attacher.destroy
   end
 
+  def check_user_quota
+    user = User.find(created_for.nil? ? created_by : created_for)
+    if (user.used_space + attachment_data['metadata']['size']) > user.available_space &&
+       !user.available_space.zero?
+      return false
+    end
+
+    true
+  rescue ActiveRecord::RecordNotFound
+    true # creating attachments without user is allowed (for tests)
+  end
+
   def attach_file
     return if file_path.nil?
     return unless File.exist?(file_path)
 
     attachment_attacher.attach(File.open(file_path, binmode: true))
     raise 'File to large' unless valid?
 
+    raise 'User quota exceeded' unless check_user_quota
```

Comment:
Rename `check_user_quota` to `user_quota_exceeded?` (and invert return values). Then this line becomes `raise 'User quota exceeded' if user_quota_exceeded?`